### PR TITLE
Remove remaining legacy template and plugin definitions

### PR DESCRIPTION
**Pull request by mfridman**
This PR removes the last of the remote gen alpha proto definitions, mainly around permissions.

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/organization.proto [hunk 1]
```diff
@@ -232,8 +232,6 @@ message GetOrganizationSettingsRequest {
 
 message GetOrganizationSettingsResponse {
   RepositoryRole repository_base_role = 1;
-  PluginRole plugin_base_role = 2 [deprecated = true];
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/organization.proto:235]: shouldn't these be reserved?

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/organization.proto:235]: The UI should be the sole consumer of these, and the underlying code has already been removed (treating this as unspecified).

We could reserve it, but given these are "internal" API contracts I'm not sure how much value there is?

Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/organization.proto:235]: Sure, I'll reserve the definitions in this PR. 

Do you want me to backfill reserved for definitions from prior PR? E.g., https://github.com/bufbuild/buf/pull/2278/files?file-filters%5B%5D=.proto&show-deleted-files=true&show-viewed-files=true#r1258876594

Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/organization.proto:235]: optimally if its quick

File: proto/buf/alpha/registry/v1alpha1/search.proto [hunk 2]
```diff
@@ -76,6 +77,9 @@ message CuratedPluginSearchResult {
 }
 
 message SearchResult {
+  reserved 5, 6;
```
Comment (mfridman) [inline proto/buf/alpha/registry/v1alpha1/search.proto:80]: Backfilled from https://github.com/bufbuild/buf/pull/2278/files?file-filters%5B%5D=.proto&show-deleted-files=true&show-viewed-files=true

File: proto/buf/alpha/registry/v1alpha1/admin.proto [hunk 3]
```diff
@@ -111,6 +111,9 @@ message ForceDeleteUserRequest {
 }
 
 message ForceDeleteUserResponse {
+  reserved 4, 5;
```
Comment (mfridman) [inline proto/buf/alpha/registry/v1alpha1/admin.proto:114]: Backfilled from https://github.com/bufbuild/buf/pull/2278/files?file-filters%5B%5D=.proto&show-deleted-files=true&show-viewed-files=true

**Review by bufdev — APPROVED** [review]

**Pull request merged**